### PR TITLE
Viz improvements

### PR DESCRIPTION
**Pull request by drewnoff**
- Added support for `extraOptions` for PlotlyChart, so now we can pass, for example, `mapboxAccessToken`.
- Upgrade plotly version.
- Created notebook with map viz examples with `CustomPlotlyChart`.
sparknotebokmapviz

### CODE REVIEW

**Comment by vidma** [comment]
test this please

File: notebooks/viz/Plotly Maps.snb [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    "source" : "## World Choropleth Map (Robinson Projection)"
+  }, {
+    "metadata" : {
+      "trusted" : true,
+      "input_collapsed" : false,
+      "collapsed" : false,
+      "id" : "E434B016D6534101A934A815873025AE"
+    },
+    "cell_type" : "code",
+    "source" : "val alcoholConsumption = scala.io.Source.fromURL(\"https://raw.githubusercontent.com/plotly/datasets/master/2010_alcohol_consumption_by_country.csv\")\n  .mkString.split(\"\\n\").drop(1)\n  .map{line => \n       val Array(location, alcohol) = line.split(\",\")\n       (location.trim, alcohol.trim.toDouble)\n      }.toSeq\n  .toDF(\"location\", \"alcohol\")",
```
Comment (vidma) [inline notebooks/viz/Plotly Maps.snb]: dumb idea, I wonder can't spark load json dataset/dataframe from https directly? would be one liner... ;)

or maybe even downloading it to local file and loading with `spark-csv` maybe would be a little cleaner too ?

Comment (drewnoff) [inline notebooks/viz/Plotly Maps.snb]: Yeah, I was wondering too about direct download to DataFrame from `http` resource, but seems like there is no such an option.
I was thinking about `/tmp/` too, but the files are not too complex yet, so decided to parse it manually. With more complex data source I would store it in intermediate file.

Comment (vidma) [inline notebooks/viz/Plotly Maps.snb]: 🆗 with me as is.

**Pull request merged**